tests/datasets/test_dataloader.py

```diff
@@ -263,3 +263,19 @@ def test_subindicators_update(self, dataset, datasetData, subindicatorGroup):
         assert len(groups) == 2
         assert groups[0].subindicators == ["A", "B", "C"]
         assert groups[1].subindicators == ["X", "Y"]
+
+    def test_new_dataset(self, dataset, datasetData, subindicatorGroup):
+        dataset = DatasetFactory()
```

Comment:
can you rename `dataset` to something else like `new_dataset` just to make it clear this is a different dataset then the fixture.